src/dmd/backend/dwarfdbginf.d

```diff
@@ -51,1532 +53,1514 @@ else
+            Section debug_loc      = { name: ".debug_loc" };
+            Section debug_abbrev   = { name: ".debug_abbrev" };
+            Section debug_info     = { name: ".debug_info" };
+            Section debug_str      = { name: ".debug_str" };
+            Section debug_line     = { name: ".debug_line" };
+        }
 
-private __gshared
-{
-
-static if (MACHOBJ)
-{
-    Section debug_pubnames = { name: "__debug_pubnames" };
-    Section debug_aranges  = { name: "__debug_aranges" };
-    Section debug_ranges   = { name: "__debug_ranges" };
-    Section debug_loc      = { name: "__debug_loc" };
-    Section debug_abbrev   = { name: "__debug_abbrev" };
-    Section debug_info     = { name: "__debug_info" };
-    Section debug_str      = { name: "__debug_str" };
-// We use S_REGULAR to make sure the linker doesn't remove this section. Needed
-// for filenames and line numbers in backtraces.
-    Section debug_line     = { name: "__debug_line", flags: S_REGULAR };
-}
-else static if (ELFOBJ)
-{
-    Section debug_pubnames = { name: ".debug_pubnames" };
-    Section debug_aranges  = { name: ".debug_aranges" };
-    Section debug_ranges   = { name: ".debug_ranges" };
-    Section debug_loc      = { name: ".debug_loc" };
-    Section debug_abbrev   = { name: ".debug_abbrev" };
-    Section debug_info     = { name: ".debug_info" };
-    Section debug_str      = { name: ".debug_str" };
-    Section debug_line     = { name: ".debug_line" };
-}
+        static if (MACHOBJ)
+            const char* debug_frame_name = "__debug_frame";
+        else static if (ELFOBJ)
+            const char* debug_frame_name = ".debug_frame";
 
-static if (MACHOBJ)
-    const char* debug_frame_name = "__debug_frame";
-else static if (ELFOBJ)
-    const char* debug_frame_name = ".debug_frame";
+        /* DWARF 7.5.3: "Each declaration begins with an unsigned LEB128 number
+         * representing the abbreviation code itself."
+         */
+        uint abbrevcode = 1;
+        AApair *abbrev_table;
+        int hasModname;    // 1 if has DW_TAG_module
 
+        // .debug_info
+        AAchars *infoFileName_table;
 
-/* DWARF 7.5.3: "Each declaration begins with an unsigned LEB128 number
- * representing the abbreviation code itself."
- */
-uint abbrevcode = 1;
-AApair *abbrev_table;
-int hasModname;    // 1 if has DW_TAG_module
+        AApair *type_table;
+        AApair *functype_table;  // not sure why this cannot be combined with type_table
+        Outbuffer *functypebuf;
 
-// .debug_info
-AAchars *infoFileName_table;
+        struct DebugInfoHeader
+        {
+          align (1):
+            uint total_length;
+            ushort version_;
+            uint abbrev_offset;
+            ubyte address_size;
+        }
 
-AApair *type_table;
-AApair *functype_table;  // not sure why this cannot be combined with type_table
-Outbuffer *functypebuf;
+        static assert(DebugInfoHeader.alignof == 1 && DebugInfoHeader.sizeof == 11);
```

Comment:
Would be good to have this in a separate commit saying that baseline compiler is now 2.076.

Reply:
Actually, a separate PR would be even better.

Reply:
I don't get it.
What does the future person who reads this line benefits with a comment saying "Baseline compiler is now 2.076" ?
I'll add the bug link, I think it's more relevant.

Doing a separate PR.

Reply:
Commit message, not comment.